### ISSUE TITLE
Remove old-style plugin authorization/GCP Login flow/Staging Plugin

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#4898]
## Feature Description

As discussed in a recent call, we internally no longer rely on the self-hosted GCP Login option/plugins to allow users to sign in to Site Kit using their own hosted GCP OAuth app.

Now that we can easily authorise a site using the Site Kit Service ("proxy"), then use `/etc/hosts` or similar locally do to local development, we should remove the ability to sign in to Site Kit in a way other than the Proxy or Staging Proxy. This will allow us to remove stories, E2E tests, components, and more to simplify the sign-in code flow.

No core team member uses this way to sign in and it's currently not well-maintained, so it makes sense to remove this.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by wpdarren** [comment]
@tofumatt @felixarntz since we will be QA'ing service updates at some stage, would removing this impact us at all? I've used it once in 15 months, so I cannot see we would but thought I'd check.

**Comment by tofumatt** [comment]
I don't think this would affect any QA. The only real reason to use this is to authorise staging sites that are not publicly available. Given that's a barely-supported feature, I don't see it affecting QA.

**Comment by aaemnnosttv** [comment]
Closing for now as I'm not sure we should do this, and the old flow has not required maintenance for a long time.
